python_modules/dagster/dagster/_config/pythonic_config/config.py

```diff
@@ -257,15 +250,15 @@ def _convert_to_config_dictionary(self) -> Mapping[str, Any]:
         Inner fields are recursively converted to dictionaries, meaning nested config objects
         or EnvVars will be converted to the appropriate dictionary representation.
         """
-        public_fields = self._get_non_none_public_field_values()
+        public_fields = self._get_non_default_public_field_values()
         return {
             k: _config_value_to_dict_representation(model_fields(self).get(k), v)
             for k, v in public_fields.items()
         }
 
-    def _get_non_none_public_field_values(self) -> Mapping[str, Any]:
+    def _get_non_default_public_field_values(self) -> Mapping[str, Any]:
```

Comment:
Here, name is changed to more accurately describe what the fn is doing - stripping out default values so that they don't accidentally trigger a validator.